### DISCUSSION TITLE
Understanding Condition tag

### DISCUSSION THREAD

**Discussion by idaamit** [wixtoolset/issues#7213] [Questions] [answered]
I have the following code : 
```
    <Condition Message="You must pass REGION_TYPE to install this product.">
            <![CDATA[Installed OR (REGION_TYPE <> "")]]>
    </Condition>
```
The message appears whenever a user didn't provide property REGION_TYPE during installation.
but I do not understand the condition `Installed OR (REGION_TYPE <> "")`
`Installed`  - means that we are in the installation flow.
`(REGION_TYPE <> "")` - means that property REGION_TYPE is not an empty string
I expected the condition to be `Installed AND (REGION_TYPE == "")`

Could you please explain why the condition works

**Comment by robmen** [comment]
> `Installed` - means that we are in the installation flow.

Uhh, no. Where did you see `Installed` documented like that?

**Comment by idaamit** [comment, marked as answer]
I understand from your reaction that I'm totally wrong. I'm confused, about how this condition works.
I'll provide a whole wxs file.
I took a small example from [https://resources.oreilly.com/examples/9781782160427/-/tree/master/Chapter%203/Built%20in%20properties%20example](WixExamples)
and added some changes : 

```
<?xml version="1.0" encoding="UTF-8"?>
<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
	<Product Id="*" Name="BuiltInPropertiesExample" Language="1033" Version="1.0.0.0" Manufacturer="WixExamples" UpgradeCode="370661cf-b3e9-4c73-bd1b-f3f6939beab4">
		<Package InstallerVersion="200" Compressed="yes" InstallScope="perMachine" />

		<MajorUpgrade DowngradeErrorMessage="A newer version of [ProductName] is already installed." />
		<MediaTemplate EmbedCab="yes" />

		<Feature Id="ProductFeature" Title="BuiltInPropertiesExample" Level="1">
			<ComponentGroupRef Id="ProductComponents" />
		</Feature>

    <Icon Id="myIcon" SourceFile="astronauta_32.ico" />
    <Condition Message="You must pass REGION_TYPE to install this product.">
      <![CDATA[Installed OR (REGION_TYPE <> "")]]>  
    </Condition>
      
	</Product>

	<Fragment>

    <SetDirectory Id="installationFolder" Value="[LOCATION]\MyConpany" Sequence="execute" >
      <![CDATA[LOCATION]]>
    </SetDirectory >
    
		<Directory Id="TARGETDIR" Name="SourceDir">
			<Directory Id="ProgramFilesFolder">
        <Directory Id="installationFolder" Name="MyConpany">
          <Directory Id="MyProduct" Name="MyProduct">
          </Directory>
        </Directory>
			</Directory>
		</Directory>
	</Fragment>

	<Fragment>
		<ComponentGroup Id="ProductComponents" Directory="MyProduct">
      <Component Id="CMP_InstallMeTXT" Guid="E8A58B7B-F031-4548-9BDD-7A6796C8460D">
        <File Id="FILE_InstallMeTXT" Source="InstallMe.txt" KeyPath="yes" />
      </Component>
		</ComponentGroup>
	</Fragment>
</Wix>
```

We get the error message during installation when we do not provide REGION_TYPE property.
Could you please explain the condition?

**Reply by robmen** [reply]
I was asking where you got this definition:

> `Installed` - means that we are in the installation flow.

because that definition is wrong and it is the source of your question. The documentation where you read it needs to be fixed, so I wanted to know where you got that definition in case it was on a WiX Toolset site.

**Reply by idaamit** [reply]
A more accurate of the definition I found  
_Installed_ - Built-in property set if product is installed locally.

![image](https://user-images.githubusercontent.com/34908206/218263561-99cc3975-3330-4676-a6d1-b6929daa7c27.png)
so, how does the condition work ?

**Reply by idaamit** [reply]
Does the message appear when the condition's value is `True` ?

**Reply by robmen** [reply]
No. [From the doc](https://wixtoolset.org/docs/v3/xsd/wix/condition/):

> Message	String	Used only under Fragment or Product elements and is required. Set the value to the text to display when the condition fails and the installation must be terminated.

**Comment by fyodorkor** [comment]
HI,
Probably here some misunderstanding of the meaning of property "Installed"
. This property is set only if the product was installed previously and you
are running the same installation for the second time. Hence you always get
your condition false. Try to change it to <![CDATA[(NOT Installed AND
REGION_TYPE) OR Installed]]>.
Try to think a bit differently - not when to show the message but when to
continue installation. According to what you wrote you want to continue in
two cases:
1. When your product has been installed before and regardless of a value of
the custom property - you pass through according to the second part of
condition
2. When you install your product for the first time and the custom property
is set - you pass through according to the first part.

On Sat, Feb 11, 2023 at 6:10 PM Rob Mensching ***@***.***>
wrote:

> No. From the doc <https://wixtoolset.org/docs/v3/xsd/wix/condition/>:
>
> Message String Used only under Fragment or Product elements and is
> required. Set the value to the text to display when the condition fails and
> the installation must be terminated.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/wixtoolset/discussions/7213#discussioncomment-4944140>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCKQ23SMFAMYYAH2BXE5XLWW62X3ANCNFSM6AAAAAAUWWQTM4>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Regards,
Fyodor Koryazhkin..
